airflow/providers/google/cloud/links/dataproc.py

```diff
@@ -71,13 +70,16 @@ def __getattr__(name: str) -> Any:
 )
 
 
-@attr.s(auto_attribs=True)
+@deprecated(
+    reason="This DataprocLink is deprecated.",
+    category=AirflowProviderDeprecationWarning,
+)
```

Comment:
Would be surprised if it not cause failure in static checks.
This class uses in some operators in top level, so this is why it pending to deprecate, without actual deprecation warnings

Reply:
You're right, it alerts the checks. I will remove it in the final version of the PR and go back to previous behaviour.